### PR TITLE
Add new error messages

### PR DESCRIPTION
**Pull request by amanda-ariyaratne**
## Purpose

This PR adds new error messages used in application sharing flows.

### CODE REVIEW

**Comment by AnuradhaSK** [comment]
Add the PR to the relevant git issue as a reference. And update the other PRs' description which is dependent on this PR. Mention those PRs should merge after this is merged.

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java [hunk 1]
```diff
@@ -474,7 +474,17 @@ public enum ErrorMessages {
...
         ERROR_CODE_ERROR_RETRIEVING_ORGANIZATION_DEPTH("65091", "Unable to retrieve the organization depth.",
-                "Server encountered an error while retrieving depth of organization with ID: %s.");
+                "Server encountered an error while retrieving depth of organization with ID: %s."),
+        ERROR_CODE_ERROR_RETRIEVING_ORGANIZATION_APPLICATIONS("65092", "Error while retrieving applications",
+                "Encountered an error while retrieving applications in organization id %s"),
+        ERROR_CODE_ERROR_CHECKING_APPLICATION_IS_A_FRAGMENT("65093", "Error while checking whether fragment app",
+                "Encountered an error while checking whether application %s is a fragment application"),
+        ERROR_CODE_INVALID_SHARE_APPLICATION_EMPTY_REQUEST_BODY("65094", "Invalid request",
+                "At least one of the attributes from shareWithAllChildren and sharedOrganizations should be present."),
+        ERROR_CODE_INVALID_SHARE_APPLICATION_REQUEST_BODY("65095", "Invalid request", "Cannot share application " +
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java:484]:
```suggestion
        ERROR_CODE_INVALID_SHARE_APPLICATION_REQUEST_BODY("65095", "Invalid request", "Cannot share the application " +
```

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java [hunk 2]
```diff
@@ -474,7 +474,17 @@ public enum ErrorMessages {
         ERROR_CODE_ERROR_FIRING_EVENTS("65090", "Error while firing the events",
                 "Server encountered an error while firing the events"),
         ERROR_CODE_ERROR_RETRIEVING_ORGANIZATION_DEPTH("65091", "Unable to retrieve the organization depth.",
-                "Server encountered an error while retrieving depth of organization with ID: %s.");
+                "Server encountered an error while retrieving depth of organization with ID: %s."),
+        ERROR_CODE_ERROR_RETRIEVING_ORGANIZATION_APPLICATIONS("65092", "Error while retrieving applications",
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java:478]: End the error messages and descriptions with a fullstop

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java (hunk 1 above)
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java:484]: These two are client errors. So the error code should start with `600` prefix. Check the client errors defined on top

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java [hunk 3]
```diff
@@ -474,7 +474,17 @@ public enum ErrorMessages {
...
+                "Encountered an error while retrieving applications in organization id %s"),
+        ERROR_CODE_ERROR_CHECKING_APPLICATION_IS_A_FRAGMENT("65093", "Error while checking whether fragment app",
+                "Encountered an error while checking whether application %s is a fragment application"),
+        ERROR_CODE_INVALID_SHARE_APPLICATION_EMPTY_REQUEST_BODY("65094", "Invalid request",
+                "At least one of the attributes from shareWithAllChildren and sharedOrganizations should be present."),
+        ERROR_CODE_INVALID_SHARE_APPLICATION_REQUEST_BODY("65095", "Invalid request", "Cannot share application " +
+                "with a set of child organizations when shareWithAllChildren is set"),
+        ERROR_CODE_ERROR_UPDATING_APPLICATION_ATTRIBUTE("65096", "Error when updating the main application",
+        "Encountered an error when updating the main application of id: %s with property share with all children.");
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java:487]: If this error is given due to the client error, move under client error enums and change the error code to `600..`

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java (hunk 2 above)
Comment (amanda-ariyaratne) [reply components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java]: Fixed with 4b1995d272dccc95fc480070cea4e2f0d297b719

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java (hunk 1 above)
Comment (amanda-ariyaratne) [reply components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java]: Fixed with 4b1995d272dccc95fc480070cea4e2f0d297b719

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java (hunk 3 above)
Comment (amanda-ariyaratne) [reply components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java]: This should belong to server errors.

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java [hunk 4]
```diff
@@ -474,7 +481,13 @@ public enum ErrorMessages {
...
                 "Server encountered an error while firing the events"),
         ERROR_CODE_ERROR_RETRIEVING_ORGANIZATION_DEPTH("65091", "Unable to retrieve the organization depth.",
-                "Server encountered an error while retrieving depth of organization with ID: %s.");
+                "Server encountered an error while retrieving depth of organization with ID: %s."),
+        ERROR_CODE_ERROR_RETRIEVING_ORGANIZATION_APPLICATIONS("65092", "Error while retrieving applications.",
+                "Encountered an error while retrieving applications in organization id %s."),
+        ERROR_CODE_ERROR_CHECKING_APPLICATION_IS_A_FRAGMENT("65093", "Error while checking whether fragment app.",
+                "Encountered an error while checking whether application %s is a fragment application."),
+        ERROR_CODE_ERROR_UPDATING_APPLICATION_ATTRIBUTE("65094", "Error when updating application.",
+        "Encountered an error when updating application of id: %s with property share with all children.");
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java:490]: check formatting. Need an indentation

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java [hunk 5]
```diff
@@ -263,6 +263,13 @@ public enum ErrorMessages {
         ERROR_CODE_UNAUTHORIZED_FRAGMENT_APP_ACCESS("60066", "Fragment applications can be managed only " +
                 "from the organization that fragment application resides.", "Application %s can't be managed " +
                 "from organization %s."),
+        ERROR_CODE_INVALID_SHARE_APPLICATION_EMPTY_REQUEST_BODY("60067", "Invalid request.",
+                "At least one of the attributes from shareWithAllChildren and sharedOrganizations should be present."),
+        ERROR_CODE_INVALID_SHARE_APPLICATION_REQUEST_BODY("60068", "Invalid request.", "Cannot share " +
+                "the application with a set of child organizations when shareWithAllChildren is set."),
+        ERROR_CODE_INVALID_DELETE_SHARE_REQUEST("60069", "Invalid request.", "Cannot delete " +
+                "sharing the application of id: %s to organization of id: %s when it is shared with all child " +
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java:271]: Shall we refactor the description here

suggestions:
1. If the app id is the fragment app id and org id is the fragment app owns org: "Cannot delete the fragment application with ID %s from the organization with ID %s if the main application is shared with all children organizations. "
2. If the app id is main app id and org id is fragment app belonging org: "Cannot unshare the application with ID %s from the organization with ID %s if the application is shared with all children organizations."

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java (hunk 4 above)
Comment (amanda-ariyaratne) [reply components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java]: Fixed with c69ad158b6ad258092ef3dc81638714283a84e98

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java (hunk 5 above)
Comment (amanda-ariyaratne) [reply components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java]: Fixed with c69ad158b6ad258092ef3dc81638714283a84e98

**Review by AnuradhaSK — APPROVED** [review]

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java [hunk 6]
```diff
@@ -263,6 +263,13 @@ public enum ErrorMessages {
         ERROR_CODE_UNAUTHORIZED_FRAGMENT_APP_ACCESS("60066", "Fragment applications can be managed only " +
                 "from the organization that fragment application resides.", "Application %s can't be managed " +
                 "from organization %s."),
+        ERROR_CODE_INVALID_SHARE_APPLICATION_EMPTY_REQUEST_BODY("60067", "Invalid request.",
+                "At least one of the attributes from shareWithAllChildren and sharedOrganizations should be present."),
+        ERROR_CODE_INVALID_SHARE_APPLICATION_REQUEST_BODY("60068", "Invalid request.", "Cannot share " +
+                "the application with a set of child organizations when shareWithAllChildren is set."),
```
Comment (Thisara-Welmilla) [inline components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java:269]: Is this happens only when `shareWithAllChildren` is set to true. If so, let's mention it like `when shareWithAllChildren is set to true`.

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java [hunk 7]
```diff
@@ -474,7 +481,13 @@ public enum ErrorMessages {
         ERROR_CODE_ERROR_FIRING_EVENTS("65090", "Error while firing the events",
                 "Server encountered an error while firing the events"),
         ERROR_CODE_ERROR_RETRIEVING_ORGANIZATION_DEPTH("65091", "Unable to retrieve the organization depth.",
-                "Server encountered an error while retrieving depth of organization with ID: %s.");
+                "Server encountered an error while retrieving depth of organization with ID: %s."),
+        ERROR_CODE_ERROR_RETRIEVING_ORGANIZATION_APPLICATIONS("65092", "Error while retrieving applications.",
+                "Encountered an error while retrieving applications in organization id %s."),
+        ERROR_CODE_ERROR_CHECKING_APPLICATION_IS_A_FRAGMENT("65093", "Error while checking whether fragment app.",
```
Comment (Thisara-Welmilla) [inline components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java:487]: `Error while checking whether fragment app` msg is not clear. Shall we have something like `Error while checking whether app is a fragment`

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java (hunk 6 above)
Comment (amanda-ariyaratne) [reply components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java:269]: resolved with 2faf5c2e64b44d7a3e1145676bd4418471d360c1

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java (hunk 7 above)
Comment (amanda-ariyaratne) [reply components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java:487]: resolved with 2faf5c2e64b44d7a3e1145676bd4418471d360c1

**Pull request merged**